packages/core/src/lib/garden/garden.ts

```
@@ -245,21 +267,32 @@ export class Garden extends EventBroker<GardenEvents> implements IGardenJS {
     if (createOrderRes.error) return Err(createOrderRes.error);
 
     const orderRes = await this.pollOrder(createOrderRes.val);
     if (orderRes.error) return Err(orderRes.error);
 
+
+    // Check if the order result is okay and if source chain is Solana, call solInit
+    if (orderRes.val && getBlockchainType(orderRes.val.source_swap.chain) === BlockchainType.Solana) {
+      const secrets = await this._secretManager.generateSecret(orderRes.val.create_order.nonce);
+      if (secrets.error) return Err(secrets.error);
+      const solInitResult = await this.solInit(orderRes.val);
+      if (solInitResult && solInitResult.error)
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)